### ISSUE TITLE
Analytic hangs when there are 0 work blocks

### ISSUE THREAD

**Issue by bentsherman** [SystemsGenetics/ACE#102]
This is a bit of an unusual case, but a valid one. It's possible to get an "empty" CCM/CMX output from similarity, where there are genes but no pairs were written to the output. You could simulate this by running similarity with mincorr = 1 and maxcorr = -1. If you then try to extract a network from the empty CCM/CMX, KINC hangs because there are no work blocks. I've traced the issue to ACE:

```
void Single::process()
{
   // While the next work index is less than the size of this object's analytic do
   // the following steps.
   while ( _nextWork < analytic()->size() )
   {
      // If this object has a simple run object then add work with a null pointer, else
      // add work with the next work block generated from this object's analytic.
      if ( _simple )
      {
         _runner->addWork(nullptr);
      }
      else
      {
         _runner->addWork(makeWork(_nextWork));
      }

      // Increment the next work index by one.
      ++_nextWork;
   }

   // ERROR: finished() is never emitted if analytic()->size() == 0
}



void SimpleRun::addWork(std::unique_ptr<EAbstractAnalyticBlock>&& block)
{
   // Save the result with a null pointer because this is simple. If the input is
   // finished then emit the finished signal.
   Q_UNUSED(block)
   _base->saveResult(nullptr);
   if ( _base->isFinished() )
   {
      emit finished();
   }
}
```

I'm getting a strong feeling of deja vu... I think it's this issue: #90. So I think you just need to apply this same fix to the Single class (and maybe others?).

**Comment by 4ctrl-alt-del** [comment]
Possible fix in commit 7465782aeaed7c9819044d9fcd07ff10e9a11ee7. Let me know if that works for you.

**Comment by bentsherman** [comment]
Fixed!
